list_updatable_packages

```diff
@@ -162,9 +168,11 @@ def build_dependency_matrix(gems: dict, project: str='foreman') -> Generator[dic
 def main() -> None:
     foreman_dependencies = list(build_dependency_matrix(gems=parse_gemfile_lock('foreman'), project='foreman'))
     katello_dependencies = list(build_dependency_matrix(gems=parse_gemfile_lock('katello'), project='katello'))
+    foreman_proxy_dependencies = list(build_dependency_matrix(gems=parse_gemfile_lock('foreman-proxy'), project='foreman'))
+    foreman_installer_dependencies = list(build_dependency_matrix(gems=parse_gemfile_lock('foreman-installer'), project='foreman'))
     plugins = list(build_plugin_matrix(find_specs()))
 
-    matrix = plugins + foreman_dependencies + katello_dependencies
+    matrix = plugins + foreman_dependencies + katello_dependencies + foreman_proxy_dependencies + foreman_installer_dependencies
```

Comment:
As a start, it could be good to have two versions showing up with PRs for a single project -- should provide a small red flag that something is amiss. Other options would be to fail the updater and hope someone notices or remove from the list of updatables any package that has a dependency conflict.

Reply:
I looked at it and it comes from the installer. For now I'd leave the installer out of it. The only dependencies from it we care about are kafo and puppet-strings.

To update things, I'd have to do a release of librarian-puppet (to get https://github.com/voxpupuli/librarian-puppet/commit/e1c4a261a93f838362d144c2d12f266369b6a13c out) and I don't have time for that right now.

So short term:
```suggestion
    matrix = plugins + foreman_dependencies + katello_dependencies + foreman_proxy_dependencies
```

Reply:
The installer dependencies identify things like clamp that as the output above shows do need an upgrade - which will be true for foreman_maintain, and hammer. Which hammer we could add in as well given it has a source release job.

I was not following the effect librarian-puppet has on these dependency calculations - can you expand?

Reply:
If you look at [`Gemfile.lock`](https://ci.theforeman.org/job/foreman-installer-develop-source-release/lastSuccessfulBuild/artifact/Gemfile.lock/*view*/) from the installer, you see:
```
    puppet_forge (2.3.4)
      faraday (>= 0.9.0, < 0.18.0, != 0.13.1)
      faraday_middleware (>= 0.9.0, < 0.15.0)
      gettext-setup (~> 0.11)
      minitar
      semantic_puppet (~> 1.0)
```

That is where this comes from:
```
    faraday (0.17.5)
      multipart-post (>= 1.2, < 3)
    faraday_middleware (0.14.0)
      faraday (>= 0.7.4, < 1.0)
```

So that ends up pulling in faraday 0.17.5. However, `puppet_forge` is not a dependency we care about in our RPM runtime. It is only a build dependency in our CI.

> The installer dependencies identify things like clamp that as the output above shows do need an upgrade - which will be true for foreman_maintain, and hammer. Which hammer we could add in as well given it has a source release job.

I'm not sure it matters. Hammer can pull in clamp too and we have nightly packages for hammer. The only thing that we care about is [this line](https://github.com/theforeman/foreman-installer/blob/b435f27e9a781d81afe8003cd7852d486770bb4b/Gemfile#L7) to pull in Kafo. We maintain Kafo ourselves so if a new release comes out, we can package it at the same time.

Actually, right now Kafo pulls in clamp with an upper bound of `< 1.3.1`. That's something we should resolve too. https://github.com/mdub/clamp/commit/4b8496d59faf14929f01689b58004e78670b8428 is something we'll need once released.

So in short: I think adding the installer dependency adds guaranteed false positives and gives no benefit.